internal/paths/glob.go

```diff
@@ -0,0 +1,91 @@
+package paths
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func ValidatePattern(pattern string) error {
+	if strings.Count(pattern, "/**/") > 1 {
+		return fmt.Errorf("the ** globbing operator may only be used once in a pattern")
+	}
+
+	// TODO: more checks
+
+	return nil
+}
+
+// Glob replaces placeholders with * and supports * and ** globbing according to docs.
+func Glob(pattern string) (matches []string, err error) {
+	pattern = filepath.Clean(pattern)
+
+	err = ValidatePattern(pattern)
+	if err != nil {
+		return nil, fmt.Errorf("invalid pattern '%s': %s", pattern, err)
+	}
+
+	if strings.Contains(pattern, "/**/") {
+		parts := strings.Split(pattern, "/**/")
+		basePattern, endPattern := parts[0], parts[1]
+
+		baseCandidates, err := filepath.Glob(basePattern)
+		if err != nil {
+			return nil, fmt.Errorf("invalid pattern '%s': %s", pattern, err)
+		}
+
+		for _, base := range directoriesOnly(baseCandidates) {
```

Comment:
👍 The globing looks much better. If this is rolled out you need to be aware of any bugsnag issues coming up. We have a lot of tests but file handling cannot be fully tested for any system. The old algorithm did "work". Just be aware and heads up that you changed it.